### DISCUSSION TITLE
Project setting to change folder location of internal folders ('shader_cache', 'logs', 'vulkan', etc)

### DISCUSSION THREAD

**Discussion by FireCatMagic** [godotengine/godot-proposals#10777] [Engine Core]
In the AppData folder there exists several folders related to internal godot things such as the shader cache. This proposal proposes adding a way to change that.
There'd be project settings (probably in advanced settings only) near the option to set a custom user:// directory. They'd have default values of "logs", "shader_cache", "vulkan" etc respectively. It would support sub directories, so "myfolder/logs" would work

**Comment by Calinou** [comment]
Out of curiosity, why do you need to change the location of these folders?

If you're looking to make an exported project portable (i.e. with all user data and cache besides the executable), https://github.com/godotengine/godot-proposals/issues/2474 would make more sense.

**Reply by FireCatMagic** [reply]
It's a silly reason, but I wanted to have those folders be saved as /godot/____ instead of in the base directory as it would be cleaner to seperate the godot files from the user files.
It certainly isn't a need, just a slight thing to help with organization

**Reply by arturopor** [reply]
> Out of curiosity, why do you need to change the location of these folders?
> 
> If you're looking to make an exported project portable (i.e. with all user data and cache besides the executable), #2474 would make more sense.

I was looking for this option, to store Godot folders out of my AppData folder. And my reason is because I have 3 hard drives and the main one, C:, is almost full so I need to keep only the necessary there (name it, the OS files and not much more).

**Reply by Calinou** [reply]
An approach like https://github.com/godotengine/godot-proposals/issues/10838 would make more sense, so you could set an environment variable and have all Godot games/apps respect it.

PS: Look into [CompactGUI](https://github.com/IridiumIO/CompactGUI) to free up some disk space :slightly_smiling_face:
